packages/manager/modules/pci/src/projects/project/additional-ips/service.js

```diff
@@ -81,13 +81,25 @@ export default class PciProjectAdditionalIpService {
     );
   }
 
-  updateInstanceForFloatingIp(projectId, region, instanceId, floatingIpId, ip) {
+  updateInstanceForFloatingIp(
+    projectId,
+    ipDetails,
+    instanceId,
+    floatingIpId,
+    ip,
+    gateway,
+  ) {
+    const { region, associatedEntity } = ipDetails;
+    const params = {
+      floatingIpId,
+      ip,
+    };
+    if (!associatedEntity?.gatewayId && gateway === null) {
+      params.gateway = null;
+    }
```

Comment:
params is a constante i'm not sure that params.gateway can modify after.

Reply:
@ghyenne if const is an object or an array we can update it, when we are adding something we are not reassigning or redeclaring it.